Add a new test with the schunk gripper which lifts a box vertically

sammy-tri opened the pull request:
Success conditions are currently disabled as the box falls from the gripper.

sammy-tri commented:
+@siyuanfeng-tri for feature review

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-:-KcOvj2TuNY415eMsY6T:blt30i2)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcOzU8RN0DyT186asHp:-KcOzU8RN0DyT186asHq:bcts5z8) (no related file):*
BTW, are there any blockers to writing a BUILD file for this stuff?  If not, BTW BTW consider doing so!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcOzU8RN0DyT186asHp:-KcP4XLfpX07g9NsxPxQ:b-1nppz9) (no related file):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW, are there any blockers to writing a BUILD file for this stuff?  If not, BTW BTW consider doing so!
</blockquote></details>

I don't think there are likely to be any blockers...  Does bazel have an equivalent construction for declaring a test "SIZE medium"?  I looked at the BUILD files for a couple of tests which have SIZE in CMakeLists.txt and bazel doesn't seem to keep that information.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: 0 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 79 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPA_p7IKgSgNgkW0It:-KcPA_p7IKgSgNgkW0Iu:bbkapi4) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L79)):*
> ```cpp
>   // Build a trajectory and PID controller for the lifting joint.
>   const auto& lifting_input_port =
>       plant->model_instance_actuator_command_input_port(0);
> ```

might as well rename 0 to lifter_instance_id?

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPChuajx6M3SEcao8c:-KcPChuajx6M3SEcao8d:b-pt8u5i) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L89)):*
> ```cpp
> 
>   auto lifting_pid_ports =
>       systems::PidControlledSystem<double>::ConnectController(
> ```

BTW, how about we change this to a const force source as well? less feedback might be better in this case. 

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 98 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPBUUrLQCkLNeaRWR0:-KcPBUUsg5maHGPtMm5P:b-5kwhgp) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L98)):*
> ```cpp
>   builder.Connect(zero_source->get_output_port(),
>                   lifting_pid_ports.control_input_port);
>   std::vector<double> lift_breaks{0., 0.9, 1., 11., 12};
> ```

11 and 12 sec? 

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 108 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPB9d3sdCWAMhb8Udt:-KcPB9d3sdCWAMhb8Udu:b-2i71ej) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L108)):*
> ```cpp
>   lift_knots.push_back(Eigen::Vector2d(1., 0.));
>   PiecewisePolynomialTrajectory lift_trajectory(
>       PiecewisePolynomial<double>::FirstOrderHold(lift_breaks, lift_knots));
> ```

i would change it to a cubic with zero start and end velocities. 

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 125 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPBe0eo_jgmuyLVVmz:-KcPBe0eo_jgmuyLVVn-:b-jhwlxt) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L125)):*
> ```cpp
>       builder.AddSystem<systems::TrajectorySource>(grip_trajectory);
>   builder.Connect(grip_source->get_output_port(),
>                   plant->model_instance_actuator_command_input_port(1));
> ```

i'd rename it to wsg instance. 

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 185 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPCDx2tNFHEolBVZ3a:-KcPCDx2tNFHEolBVZ3b:b83a1gs) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L185)):*
> ```cpp
>   // Expect that the gripper is open (even though clamping force is
>   // being applied).
>   EXPECT_GT(final_output_data[
> ```

BTW, might want to check the pose of the box instead of just z. 

---

*[drake/examples/schunk_wsg/test/test_lifter.sdf, line 47 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPCSwaFtgMsznJrXpZ:-KcPCSwaFtgMsznJrXp_:bmikxu0) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/test_lifter.sdf#L47)):*
> ```
>         </dynamics>
>       </axis>
>       <physics>
> ```

BTW, i think we can delete the following. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Review status: 0 of 4 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcOzU8RN0DyT186asHp:-KcPECZsdYhoSg6P_T78:b3lr8ar) (no related file):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I don't think there are likely to be any blockers...  Does bazel have an equivalent construction for declaring a test "SIZE medium"?  I looked at the BUILD files for a couple of tests which have SIZE in CMakeLists.txt and bazel doesn't seem to keep that information.
</blockquote></details>

https://bazel.build/versions/master/docs/be/common-definitions.html#test.size

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Review status: 0 of 6 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcOzU8RN0DyT186asHp:-KcPNguK3PXlMmq9N5dI:b-qx4ut6) (no related file):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

https://bazel.build/versions/master/docs/be/common-definitions.html#test.size
</blockquote></details>

Done (skipped test size part since we don't CI bazel debug tests AFAICT)

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 79 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPA_p7IKgSgNgkW0It:-KcPPvbNGP_Nvd7l8fEd:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L79)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

might as well rename 0 to lifter_instance_id?
</blockquote></details>

Done.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPChuajx6M3SEcao8c:-KcPPwLaVAqELKQfjQ7i:be53xy5) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L89)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

BTW, how about we change this to a const force source as well? less feedback might be better in this case. 
</blockquote></details>

That was my initial implementation.  I didn't really like the results, mainly that in order to leave the ground without waiting forever due to the small net acceleration, the gripper would hit the limit pretty hard at the top of the prismatic joint.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 98 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPBUUrLQCkLNeaRWR0:-KcPRAQSpGaiP_0Qb6Lb:b-bizqh8) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L98)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

11 and 12 sec? 
</blockquote></details>

Added comment.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 108 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPB9d3sdCWAMhb8Udt:-KcPRKpWWpSbGWCYYFNn:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L108)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

i would change it to a cubic with zero start and end velocities. 
</blockquote></details>

Done.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 125 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPBe0eo_jgmuyLVVmz:-KcPRMp34W6lDZaBtn9a:b-o4vviu) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L125)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

i'd rename it to wsg instance. 
</blockquote></details>

Added instance id instead of magic "1"...  or did you mean something else?

---

*[drake/examples/schunk_wsg/test/test_lifter.sdf, line 47 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPCSwaFtgMsznJrXpZ:-KcPRoNzWZVdr4mCTArM:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/test_lifter.sdf#L47)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

BTW, i think we can delete the following. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPChuajx6M3SEcao8c:-KcPTFMvkp8mWaHz2noa:bkk1xmy) ([raw file](https://github.com/robotlocomotion/drake/blob/aeedc6160b2111bb2943b6836279dd793ff0f25e/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L89)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

That was my initial implementation.  I didn't really like the results, mainly that in order to leave the ground without waiting forever due to the small net acceleration, the gripper would hit the limit pretty hard at the top of the prismatic joint.
</blockquote></details>

BTW, fair enough. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-:-KcPTli8PUWSRPPZ4Nwn:bnfp4nl)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
+@david-german-tri for platform review

---

Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-:-KcPVvfiFng10TBG-xaA:b-rhk18b)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 4 files at r1, 3 of 5 files at r2.
Review status: 4 of 6 files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/examples/schunk_wsg/BUILD, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPXnGvwdERl3E4kDh4:-KcPXnGvwdERl3E4kDh5:b-c4id7s) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/BUILD#L37)):*
> ```Python
> # === test/ ===
> 
> drake_cc_googletest(
> ```

BTW if this requires SNOPT please add `tags = ["snopt"]`

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 128 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPbY9lVr1My0asu2Xo:-KcPbY9mAgvNL6V7H-Hf:bqcs0h2) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L128)):*
> ```cpp
>       builder.AddSystem<systems::TrajectorySource>(lift_trajectory);
>   builder.Connect(lift_source->get_output_port(),
>                   lifting_pid_ports.state_input_port);
> ```

BTW, this name is really confusing: I had to do a lot of digging to determine it's the _desired_ state, and that the actual state is observed through connections that aren't visible in this file.  No change needed, just griping.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 172 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPbrZ2Zi2rzVnHex7x:-KcPbrZ2Zi2rzVnHex7y:bxbj7ii) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L172)):*
> ```cpp
>   auto positions = tree.computePositionNameToIndexMap();
>   ASSERT_EQ(positions["left_finger_sliding_joint"], 1);
>   plant_initial_state(1) = -0.0550667;
> ```

BTW this is pretty magical.  Should it live in a helper function somewhere?  Or at least have a comment explaining what it means?

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 198 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPc-Rcf1b9NsFOOPHh:-KcPc-Rcf1b9NsFOOPHi:b-m8auk8) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L198)):*
> ```cpp
> 
>   // TODO(sam.creasey) Re-enable the tests below once the gripper can
>   // actually hold the box.
> ```

BTW, that will be awesome, but in the meantime consider not leaving an `#if 0` hanging around.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-:-KcPcB2pESMgHQm4rMVB:bnfp4nl)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Reviewed 2 of 5 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/schunk_wsg/BUILD, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPXnGvwdERl3E4kDh4:-KcPijD9lpeqXcudPAEm:b-rfn9j3) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/BUILD#L37)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW if this requires SNOPT please add `tags = ["snopt"]`
</blockquote></details>

I don't believe that it does.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 172 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPbrZ2Zi2rzVnHex7x:-KcPjYCK8r1hqpRbwHmI:b3s3nru) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L172)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW this is pretty magical.  Should it live in a helper function somewhere?  Or at least have a comment explaining what it means?
</blockquote></details>

Done. (commented)

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 198 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPc-Rcf1b9NsFOOPHh:-KcPjaLg16JaUJc2QXfy:b-u1f20n) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L198)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW, that will be awesome, but in the meantime consider not leaving an `#if 0` hanging around.
</blockquote></details>

Would commenting out the lines of code be better?  Or a comment explaining what the target values are and no dead code?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc, line 198 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5090#-KcPc-Rcf1b9NsFOOPHh:-KcPtDxcYi4PrxmwSHeH:bj1gm71) ([raw file](https://github.com/robotlocomotion/drake/blob/fecc22745c1997e9fd9f4d02c7f5dde9b99bb4f9/drake/examples/schunk_wsg/test/schunk_wsg_lift_test.cc#L198)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Would commenting out the lines of code be better?  Or a comment explaining what the target values are and no dead code?
</blockquote></details>

The latter would be better IMO, but if you don't want to go through a CI redo just for that I understand.  Marking myself resolved.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5090)*
<!-- Sent from Reviewable.io -->
